packages/release-brancher/test/release-brancher.ts

```diff
@@ -0,0 +1,336 @@
+        assert.ok(newConfig);
+        snapshot(newConfig);
+      });
+    });
+    describe('with releaseType', () => {
+      beforeEach(() => {
+        runner = new Runner({
+          branchName: '1.x',
+          targetTag: 'v1.3.0',
+          gitHubToken: 'sometoken',
+          upstreamOwner: 'testOwner',
+          upstreamRepo: 'testRepo',
+          releaseType: 'custom-releaser',
+        });
+      });
+      it('updates a basic config', async () => {
+        const config = loadFixture('release-please/basic.yaml');
+        const newConfig = runner.updateReleasePleaseConfig(config);
+        assert.ok(newConfig);
+        snapshot(newConfig);
+      });
+  
+      it('updates a config with extra branches already configured', async () => {
+        const config = loadFixture('release-please/with-extra-branches.yaml');
+        const newConfig = runner.updateReleasePleaseConfig(config);
+        assert.ok(newConfig);
+        snapshot(newConfig);
+      });
+    });
+    it('ignores branch if it already exists', () => {
+      const config = loadFixture('release-please/with-extra-branches.yaml');
+      runner = new Runner({
+        branchName: '3.1.x',
+        targetTag: 'v3.1.0',
+        gitHubToken: 'sometoken',
+        upstreamOwner: 'testOwner',
+        upstreamRepo: 'testRepo',
+      });
+      const newConfig = runner.updateReleasePleaseConfig(config);
+      assert.equal(newConfig, undefined);
+    });
+  });
+
+  describe('updateSyncRepoSettings', () => {
+    it('updates a basic config', async () => {
+      const config = loadFixture('sync-repo-settings/basic.yaml');
+      runner = new Runner({
+        branchName: '1.x',
+        targetTag: 'v1.3.0',
+        gitHubToken: 'sometoken',
+        upstreamOwner: 'testOwner',
+        upstreamRepo: 'testRepo',
+      });
+      const newConfig = runner.updateSyncRepoSettings(config);
+      assert.ok(newConfig);
+      snapshot(newConfig);
+    });
+
+    it('updates a config with extra branches already configured', async () => {
+      const config = loadFixture('sync-repo-settings/with-extra-branches.yaml');
+      runner = new Runner({
+        branchName: '1.x',
+        targetTag: 'v1.3.0',
+        gitHubToken: 'sometoken',
+        upstreamOwner: 'testOwner',
+        upstreamRepo: 'testRepo',
+      });
+      const newConfig = runner.updateSyncRepoSettings(config);
+      assert.ok(newConfig);
+      snapshot(newConfig);
+    });
+
+    it('ignores branch if it already exists', () => {
+      const config = loadFixture('sync-repo-settings/with-extra-branches.yaml');
+      runner = new Runner({
+        branchName: '3.1.x',
+        targetTag: 'v3.1.0',
+        gitHubToken: 'sometoken',
+        upstreamOwner: 'testOwner',
+        upstreamRepo: 'testRepo',
+      });
+      const newConfig = runner.updateSyncRepoSettings(config);
+      assert.equal(newConfig, undefined);
+    });
+
+    it('ignores branch if repo has no branch protection enabled', () => {
```

Comment:
A little confused by this assertion, don't we want to enable branch protection if it's not set (based on your comment in the issue body).

Reply:
The way it works is that it's trying to duplicate the branch protection rules for a known branch rather than having you specify it when you run this command.

An alternative would be to throw an error in this case as we don't know what branch protection rules to create.

Reply:
Updated to throw instead. We can revisit if we expect sync-repo-settings configs to have no configured branch protection rules in the future.